src/services/invokeService.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseService } from "./baseService"
+import Serverless from "serverless";
+import config from "../config";
+import axios from "axios";
+import { FunctionAppService } from "./functionAppService";
+import { FunctionAppResource } from "../armTemplates/resources/functionApp";
+import { ServerlessAzureConfig } from "../models/serverless";
+
+export class InvokeService extends BaseService {
+  public serverless: Serverless;
+  public options: Serverless.Options;
+  public constructor(serverless: Serverless, options: Serverless.Options) {
+    super(serverless, options);
+    this.serverless = serverless;
+    this.options = options;
+  }
+
+  /**
+   * Invoke an Azure Function
+   * @param functionName Name of function to invoke
+   * @param data Data to use as body or query params
+   * @param method GET or POST
+   */
+  public async invoke(functionName: string, data: any, method: string){
+    /* accesses the admin key */
+    if (!(functionName in this.slsFunctions())) {
+      this.serverless.cli.log(`Function ${functionName} does not exist`);
+      return;
+    }
+    const functionObject = this.slsFunctions()[functionName];
```

Comment:
You are already making call to `slsFunctions()` above.  Consider caching that result and reusing here to avoid the duplicate call.

Reply:
This still calls the function 2x.  Consider caching the result of the function call and resusing the variable.